### PR TITLE
Add blast+

### PR DESCRIPTION
**Pull request by colinbrislawn**
Addresses #19 

First recipe from precomplied binaries. I will probably have to refine it. Suggestions welcome!

### CODE REVIEW

**Comment by colinbrislawn** [comment]
Can someone help me out with this?

I'm trying to use `$TRAVIS_OS_NAME` to identify the OS of this travis build in order to download and deploy the correct binary, but I can't get this variable to work.

**Comment by mortonjt** [comment]
It looks like you aren't the first one having trouble setting `$TRAVIS_OS_NAME`

http://stackoverflow.com/questions/32142004/usage-of-travis-os-name-in-travis-yml
https://github.com/travis-ci/travis-ci/issues/2205
https://github.com/travis-ci/travis-ci/issues/4627

Personally, I'm not sure about how to fix this issue.  But a sure way around this is to just use `uname`.

**Comment by Jorge-C** [comment]
In the `meta.yml` it's possible to restrict certain lines to be executed in different OSes (so called [selectors](http://conda.pydata.org/docs/building/meta-yaml.html#preprocessing-selectors)), and then you can use the [`build: script`](http://conda.pydata.org/docs/building/meta-yaml.html#script) section to have one script for linux and another one for OSX.

There should never be references to travis env variables in build scripts, since that would make it impossible to build them in "normal" computers.

**Comment by RNAer** [comment]
@colinbrislawn , please use `uname`. If Travis env variable is used in the recipe - even if it works, it is buildable only on Travis.

**Comment by colinbrislawn** [comment]
@RNAer, I like the idea of using `uname` so that it would work independently of Travis.
I also like the idea of using selectors + build: script to ditch build.sh entirely as mentioned by @Jorge-C.

Any advice here? Which method may be better for getting precomplied binaries?

**Comment by Jorge-C** [comment]
Please use selectors, which is the way to do this in conda recipes. Don't try to reimplement them using `uname`!

**Comment by colinbrislawn** [comment]
Point taken! I appreciate your guidance with conda.

**Comment by RNAer** [comment]
right, selector is better.

**Comment by colinbrislawn** [comment]
I'm still struggling with this one.

I can download and extract the right binary for the OS using the method @Jorge-C mentioned.

I'm unable to copy all the excitable files into the proper `$PREFIX/bin` folder. Does anyone on this team know how to locate (or change!) the location of the decompressed source tree that is stored during a conda build?

```
recipes bris469$ conda build blastplus/
BUILD START: blast+-2.2.31-2
Source cache directory is: /Users/bris469/anaconda/conda-bld/src_cache
Found source in cache: ncbi-blast-2.2.31+-universal-macosx.tar.gz
Package: blast+-2.2.31-2
source tree in: /Users/bris469/anaconda/conda-bld/work/ncbi-blast-2.2.31+
```

If I could get a last line, I finish this process.

**Comment by Jorge-C** [comment]
Would the `SRC_DIR` env variable work for you? See the available ones
[here](http://conda.pydata.org/docs/building/environment-vars.html).

2015-11-09 17:58 GMT-07:00 Colin Brislawn notifications@github.com:

**Comment by colinbrislawn** [comment]
Oh I think that's perfect! It just built on my machine, so let's see how Travis likes it.

**Comment by colinbrislawn** [comment]
It works! I'm not sure if we should merge yet. I still have to:
- test `build.bat` on Windows
- clean up `build.sh`
- add tests for more then one blast executable file

**Comment by Jorge-C** [comment]
Great! Let us know when you consider it ready or if you have any other questions. Thanks for your help!

File: recipes/blastplus/build.bat [hunk 1]
```diff
@@ -0,0 +1,2 @@
+bash configure --prefix=%PREFIX%
+xcopy /s "ncbi-blast-2.2.31+\bin\*.exe" "%PREFIX%\bin"
```
Comment (RNAer) [inline recipes/blastplus/build.bat]: missing a newline

File: recipes/blastplus/build.sh [hunk 2]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+
+mkdir -p "$PREFIX/bin"
+
+cd $SRC_DIR
+ls -alh
```
Comment (RNAer) [inline recipes/blastplus/build.sh]: lines 5-6 are not needed?

File: recipes/blastplus/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  md5: ae789a5bfb523cef95ac40e5a6b1929e                                                                          # [linux64]
+  fn: ncbi-blast-2.2.31+-universal-macosx.tar.gz                                                                 # [osx]
+  url: ftp://ftp.ncbi.nlm.nih.gov/blast/executables/blast+/2.2.31/ncbi-blast-2.2.31+-universal-macosx.tar.gz     # [osx]
+  md5: 316555fef19749fe90455b7def03634e                                                                          # [osx]
+  fn: ncbi-blast-2.2.31+-x64-win64.tar.gz                                                                        # [win64]
+  url: ftp://ftp.ncbi.nlm.nih.gov/blast/executables/blast+/2.2.31/ncbi-blast-2.2.31+-x64-win64.tar.gz            # [win64]
+  md5: f80491198c16daa2d82ab3c95d00f244                                                                          # [win64]
+
+build:
+  number: 1
```
Comment (RNAer) [inline recipes/blastplus/meta.yaml]: The build number starts with zero. so better to use zero here.

**Comment by RNAer** [comment]
looks good except a few minor things.

**Comment by colinbrislawn** [comment]
If this builds, I think it's ready to merge. I can address any lingering Windows problems when we address #13 and we can add more tests as needed.

**Comment by Jorge-C** [comment]
Thanks! Do you mind squashing your commits? That way it helps keep all master commits building, which can eg ease using bisect later on.

**Comment by colinbrislawn** [comment]
Sure thing! I'll do this later this afternoon.

**Comment by colinbrislawn** [comment]
There may be a better way to squash those... recommendations are welcome.

**Comment by mortonjt** [comment]
Try `git rebase -i 32d8be3`

**Comment by colinbrislawn** [comment]
OK, so I should squash everything all the way back to `32d8be3` `ENH: added sortmerna`? I have squashed mine, but unintentionally rebased over the top of others commits. Sigh, [git is hard](https://xkcd.com/1597/).

**Comment by mortonjt** [comment]
Oh, I mixed your commit with @wasade 's commit :S

No you don't want to do that.

**Comment by mortonjt** [comment]
Yeah, the commit history looks screwy.  Perhaps checking out a new branch from master, incorporate your changes and issue another PR may be the most seamless approach.

**Comment by colinbrislawn** [comment]
I think I fixed the extraneous commits. If this looks ok, I believe it's ready to merge.

**Comment by Jorge-C** [comment]
Thank you Colin!

**Pull request merged**